pennylane/templates/state_preparations/cosine_window.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the CosineWindow template.
+"""
+import numpy as np
+import pennylane as qml
+from pennylane.operation import AnyWires, StatePrepBase
+from pennylane.wires import Wires, WireError
+from pennylane import math
+
+
+class CosineWindow(StatePrepBase):
+    r"""CosineWindow(wires)
+    Prepare the initial state following a cosine wave function.
+
+    .. math::
+
+        |\psi\rangle = \sqrt{2^{1-m}} \sum_{k=0}^{2^m-1} \cos(\frac{\pi k}{2^m} - \frac{\pi}{2}) |k\rangle,
+
+    where :math:`m` is the number of wires.
+
+    .. figure:: ../../_static/templates/state_preparations/cosine_window.png
+        :align: center
+        :width: 65%
+        :target: javascript:void(0);
+
+    .. note::
+
+        The wave function is shifted by :math:`\frac{\pi}{2}` units so that the window is centered.
```

Comment:
Very minor but could the shift be user controlled?

Reply:
The result would be this, which is not exactly the cosine since it is truncated and may confuse the user more. Let's leave it like this for the moment.

<img width="512" alt="Captura de pantalla 2023-10-17 a las 14 09 57" src="https://github.com/PennyLaneAI/pennylane/assets/65235481/be23b29e-037d-4f3b-894c-59278da70c50">